### PR TITLE
Implement typeguarding with the typia library

### PR DESCRIPTION
**Pull request by jr1221**
This uses the [typia](https://typia.io/docs/) compiletime type validation system to check the types of `SubcriptionMessage` and `ServerData` from `Odyssey-Base`.  It is currently in generation mode as bun seems to be a nonstandard typescript compiler.  Code generation should work for any future customizations of the above types.  

It works by reading the typia/templates files at build time and outputting the typia/generated.  The templates file takes the types from `Odyssey-Base`, re-exports them so they can be accessed in the same import as the typeguards, and creates a bunch of `const` functions that provide the type validation.  There are various types of such functions documented at the link above, basic comments are included with each one.  It is important we don't use `typia.is` imported from typia to type check as that is not size efficient.  Also we can eventually comment the ones we dont use out to reduce the binary size, which is about 700 lines right now.  Only using the `checkIs` for each type would only be about 40-50 lines.

Other features:
it can stringify (implemented) and parse (not implemented) more efficiently than the JSON namespace, so for publishing we may want to use that stringify.  
it can generate random objects, which may be useful for testing libraries.  
it can deep clone objects (not currently implemented).

This PR switches the `message` body to harness `checkIs`, but we may want to switch to validation to get good errors.

### CODE REVIEW

**Review by evanlombardo — CHANGES_REQUESTED** [review]
Great job, this is an amazing addition so I'm glad it worked out. Two quick things, see my other comment and also add typia's installation to the `install` script.

File: siren/typia/templates/message.types.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import typia from "typia";
+import { SubscriptionMessage, ServerMessage } from "../../Odyssey-Base/src/types/message.types";
+
+export type { SubscriptionMessage, ServerMessage };
+
```
[the remark this answers is not in the document]
Comment (jr1221) [reply siren/typia/templates/message.types.ts:5]: The `templates` directory is what I wrote to tell typia to generate validation, which goes in the `generated` directory which I did add to the gitignore.  So the file above is basically not used at runtime but still needed for compile time.

**Comment by jr1221** [comment]
>  also add typia's installation to the `install` script.
Should I just add `bun install` to the `install` script?  Because that should pull in everything listed in `package.json` which includes `bun-types` and all of `typia` and its dependencies.

**Comment by evanlombardo** [comment]
> > also add typia's installation to the `install` script.
> 
> Should I just add `bun install` to the `install` script? Because that should pull in everything listed in `package.json` which includes `bun-types` and all of `typia` and its dependencies.
> 
> Edit: [link to docs](https://bun.sh/docs/cli/install#bun-install)

Sounds good

**Comment by jr1221** [comment]
For some reason that doesn't work for me.  I committed it nonetheless maybe its a problem with my machine.  We could always look at it later.

**Review by evanlombardo — APPROVED** [review]

**Pull request merged**